mqtt/mqtt-bridge/src/client.rs

```diff
@@ -194,20 +318,35 @@ impl<T: EventHandler> MqttClient<T> {
             .await
             .map_err(ClientConnectError::PollClientFailure)?
         {
-            //TODO: change the mqtt client to send an error back when the subscriotion fails instead of reconnecting and resending the sub
-            //right now it can't detect if the the broker doesn't allow to subscribe to the specific topic, but it will send a connect event
-            if let Event::NewConnection { reset_session: _ } = event {
-                return Ok(());
+            if let Event::SubscriptionUpdates(subscriptions) = event {
+                for subscription in subscriptions {
+                    match subscription {
```

Comment:
Is it guaranteed that .try_next() will return the SubscriptionUpdates and not any other event for already running client? Also, I'm not sure if this loop will terminate properly.

Reply:
It can be that it returns something else, but the subscription awaits so there should be no other events in the meantime. I think it can be consolidated when Andrew has the changes for the pump.